### ISSUE TITLE
Locale from `checkoutConfiguration` not applied

### ISSUE THREAD

**Issue by markosopcic** [Adyen/adyen-android#1558]
Hi everyone. 
I'm implementing Adyen in Compose. I wanted to implement locale switching based on my app config, but it doesn't seem to work. 
Here's the code I use to initialize Adyen:

```kotlin
LaunchedEffect(order) {
        coroutineScope.launch {
            val result = CheckoutSessionProvider.createSession(
                sessionModel = SessionModel(adyenSessionData.sessionId, adyenSessionData.sessionData),
                environment = environment,
                clientKey = adyenSessionData.config.clientKey
            )
            when (result) {
                is CheckoutSessionResult.Error -> onPaymentFailed()
                is CheckoutSessionResult.Success -> DropIn.startPayment(
                    context = context,
                    dropInLauncher = dropInLauncher,
                    checkoutSession = result.checkoutSession,
                    checkoutConfiguration = CheckoutConfiguration(
                        environment,
                        adyenSessionData.config.clientKey,
                        Locale.forLanguageTag("nl-NL")
                    )
                )
            }
        }
    }
```

As you can see I'm setting the locale in the `CheckoutConfiguration` object. 

I've tried setting the `Locale.forLanguageTag("nl-NL")`, `Locale.GERMAN`, `Locale.GERMANY`, but nothing seems to work, the DropIn component still opens up in english. 
<img width="225" alt="image" src="https://github.com/Adyen/adyen-android/assets/32719316/eba23b3c-e112-4028-a78e-f6003aa62266">

I'm testing on the test environment.

Please let me know if I can provide any more information regarding this issue to help speed up the resolution, thanks.

**Comment by jreij** [comment]
Hi @markosopcic, thanks for reaching out. When using drop-in, the payment methods list is fetched and localized on the Adyen backend, which is why you need to set the [shopperLocale in the /sessions request](https://docs.adyen.com/api-explorer/Checkout/latest/post/sessions#request-shopperLocale) on your backend (I assume you need to pass this from your mobile app first, unless you have your user preferences saved on your backend as well). After doing that you don't need to set the shopper locale again in the `CheckoutConfiguration`, you can omit that parameter altogether.

However, it looks strange to me that the `PAYMENT METHODS` label on top of your screenshot is also in English, the value for this label is stored and localized on our side in the SDK so it should have shown in Dutch or German. But for now let's try out the first solution and see if it works.

**Comment by jreij** [comment]
Hi @gabrielglbh, thanks for reporting this as well. Just to be sure, you and @markosopcic are not working on the same app right? 😄 

I have a few questions for both of you:
- Are there any strings at all that are showing up in Spanish? Can you share a few screenshots of your drop-in? For example the payment methods list screen and the card component screen?
- Can you check whether you have `resConfigs` or any resource related configuration in your gradle files?
- Can you share the code where you create the `CheckoutConfiguration` and also the code where you call `DropIn.startPayment`?

**Comment by jreij** [comment]
@gabrielglbh thanks for the info. I have a few more questions:
- Do you get the same result if you manually set the shopper locale when creating the `CheckoutConfiguration`, like this:
```kotlin
CheckoutConfiguration(
            environment = environment,
            clientKey = clientKey,
            shopperLocale = Locale("es", "ES"),
)
```
This should not be required indeed but I'm just double checking a few things now.
- What is the device language? Does changing it have any effect on this?
- Which version are you using? Could you try to update to `5.3.1` if you are not there yet?

**Comment by gabrielglbh** [comment]
@jreij Thank you for the fast answers! 

Yes indeed, putting (or not) the locale in the configuration does give the same result: always in english.

- The device language has been set to english and spanish giving the same result: always english.
- And, sadly yes, I am using `5.3.1`. 

I think I am not helping at all 🥲

**Comment by jreij** [comment]
@gabrielglbh quite the opposite, you're helping a lot and quickly 😄 Unfortunately I cannot reproduce yet.

Can you try to log the value of `getString(R.string.checkout_card_number_hint)` while changing your device language? If you always get `Card number` even with other languages then we know that the translated resources are just completely being removed on build for some reason. Otherwise we'll continue investigating other areas.

**Comment by gabrielglbh** [comment]
@jreij Quick update. It seems we DO have `resConfigs` like this ~~it was very well hidden~~:

```kotlin
resourceConfigurations += brandConfiguration.languageResourcesConfiguration.map { it.language }
```

The `brandConfiguration.languageResourcesConfiguration` contains a list of `Locale` like `listOf(Locale("en", "US"), Locale("es", "ES"))` or `listOf(Locale("es", "ES"))`.

When removing that line from the gradle file, the DropIn actually translates. You had it right at the beginning!! Thank you!

My team is reporting that the `resConfig` worked well along previous adyen versions. Out of curiosity, why does the `resConfig` conflict with the translation in version `5.x`?

**Comment by jreij** [comment]
@gabrielglbh glad to hear that.

> why does the resConfig conflict with the translation in version 5.x?

I don't think we changed anything in v5 that affects this. However, looking at your code I can see that `brandConfiguration.languageResourcesConfiguration.map { it.language }` will translate into a list of languages like `'de', 'es', 'nl'...`. But Android requires this attribute to exactly match the values we have on our side so it should be `'de-rDE', 'es-rES', 'nl-rNL'...` because we include the region as well.

Check the full list [here](https://github.com/Adyen/adyen-android/tree/develop/drop-in/src/main/res) or in the screenshot below. Let me know if that works.

![Screenshot 2024-04-17 at 17 10 15](https://github.com/Adyen/adyen-android/assets/9079915/6d4bc2b4-3cc4-414b-b29f-5dcdb27bb970)

**Comment by gabrielglbh** [comment]
@jreij That works. Thank you very much for the help, appreciate it! Now the DropIn translates at will.

**Comment by markosopcic** [comment]
My issue might be connected, however I am not experienced in `resConfigs`, I'd appreciate any assistance.  We have the following `resConfigs` defined:

```    
def RES_CONFIGS = [
        'ar', 'bg', 'cs', 'da', 'de', 'el', 'en', 'es', 'et', 'fi', 'fr', 'hi', 'he', 'hu', 'in', 'it', 'iw', 'ko', 'nb', 'lt', 'nl', 'pl',
        'pt', 'ro', 'ru', 'sk', 'sl', 'sv', 'th', 'tr', 'uk', 'vi', 'zh', 'hr', 'lv', 'sr',

        'ar-rBH', 'ar-rKW', 'ar-rSA', 'ar-rQA', 'bg-rBG', 'cs-rCZ', 'de-rCH', 'fr-rBE', 'fr-rCA', 'hu-rHU', 'en-rAU', 'en-rGB', 'en-rIN',
        'en-rNZ', 'en-rUS', 'en-rZa', 'es-rMX', 'es-rUS', 'in-rID', 'ko-rKR', 'nl-rBE', 'pt-rBR', 'pt-rPT', 'ru-rUA', 'ru-rRU', 'ro-rRO',
        'th-rTH', 'tr-rTR', 'uk-rUA', 'zh-rCN', 'zh-rHK', 'zh-rTW', 'es-rAR', 'es-rCL', 'ar-rOM', 'en-rPH', 'el-rGR', 'sk-rSK', 'sl-rSI',
        'lt-rLT', 'et-rEE', 'he-rIL', 'iw-rIL', 'hr-rHR', 'lv-rLV', 'ar-rJO', 'fr-rMU', 'zh-rMO', 'el-rCY', 'sr-rRS', 'hi-rIN'
    ] as String[]
...

defaultConfig {
        resConfigs(*RES_CONFIGS)
}
```

From my understanding, `resConfigs` defines the resource quantifiers which *will not* be filtered out of the packaged app. For example, in the list we have the config  `nl-rBE` - when i set the locale to `nl-BE`, and log the string value you suggested above, i get the english version. I see there is no string qualifier nl-rBE in the adyen resources, so I added the following res_configs (as we need the adyen SDK to work in the Netherlands and Belgium):
`'nl-rNL', 'en-rNL', 'en-rBE'`
After this I run the app, I set the Locale to `nl-BE`, and I log the same string, and now it is in Dutch. 
Although, when I run the adyen component, it is still in english, unless i set the `shopperLocale` manually. Is this expected?

Also, when I set it to `fr-BE`, both the log and SDK are in english, even though we have the `fr-rBE` res config in the list.

**Comment by jreij** [comment]
> From my understanding, resConfigs defines the resource quantifiers which will not be filtered out of the packaged app

@markosopcic this is correct. So the values you have inside `resConfigs` have to include the values we defined on our side, take a look at the screenshot below. Therefore as @gabrielglbh suggested, you need to add `nl-rNL` and `fr-rFR` to your `resConfigs`. And you will still be able to use `nl-BE` and `fr-BE` as your shopper locale after that.

![Screenshot 2024-04-17 at 17 10 15](https://github.com/Adyen/adyen-android/assets/9079915/6d4bc2b4-3cc4-414b-b29f-5dcdb27bb970)

> Although, when I run the adyen component, it is still in english, unless i set the shopperLocale manually. Is this expected?

This might be a different issue. You shouldn't need to set shopper locale in the `CheckoutConfiguration` if you are passing it to the `/sessions` API call on your backend. Can you try this and the `resConfigs` solution above then let us know? If it still doesn't work please share screenshots of the payment methods list screen and the card component screen.

**Comment by markosopcic** [comment]
Thanks. Unfortunately, our backend team can't prioritise adding the locale to the `/sessions` call right now, but I believe that it will resolve our issue.
